### PR TITLE
Remake template to `gem_generator` compatible

### PR DESCRIPTION
**Pull request by AlexWayfer**
Hi again!

As we [discussed here](https://github.com/lostisland/faraday-middleware-template/pull/1#issuecomment-924652437), you're not against the idea to use some generator for template instead of manual replaces and checks.

So, here we are! I've completed `--git` option for `gem_generator`, tested it completely, and ready to introduce a version of your template, but with variables (and file names) for `gem_generator`.

(even with self-specs how it works)

What do you think?

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  rubocop:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 3.0
```
Comment (olleolleolle) [inline .github/workflows/ci.yaml]: Super-tiny, not important: quote it as "3.0" here, too.

File: README.md [hunk 2]
```diff
@@ -8,37 +8,48 @@ These classes are called middleware. Use this repository to create your own midd
...
-The next step is for you to find and replace all the "parametrised" names in this template
-and change them to make it unique.
+The next step is for you to generate a new Faraday middleware via `gem_generator` call.
 
 First of all, you should decide on the name of your middleware.
 The current convention (which is by no means mandatory) is to call middleware gems as `faraday-<something>`.
 Here are some examples:
 
 * `Hashie::Mash::Rash`: [`faraday-rashify`][faraday-rashify]
+* XML (request): [`faraday-encode_xml`][faraday-encode_xml]
```
Comment (olleolleolle) [inline README.md:28]: Super to have something with two words. 🏆

File: template/.github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+jobs:
+  rubocop:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 3.0
```
Comment (olleolleolle) [inline template/.github/workflows/ci.yaml:15]:
```suggestion
        ruby-version: '3.0'
```

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (AlexWayfer) [reply .github/workflows/ci.yaml]: Seems working, but fair enough. Updated.

**Comment by AlexWayfer** [comment]
I don't know why GitHub Actions are not triggered. I've checked: `.yaml` files are supported. Guess because I have no rights for this repository, will see, I ready to fix if there will be problems.

Also a note: "Use this template" button should be disabled in case of merging this PR.

image

**Review by olleolleolle — APPROVED** [review]
I have now re-read this, and I'm 👍 for it.

Let's get this merged! ⛵

**Comment by olleolleolle** [comment]
I unchecked Template Repo:

bild

**Comment by olleolleolle** [comment]
@AlexWayfer Now we can see some of the Checks failing, which is good, more visibility.

**Comment by AlexWayfer** [comment]
> @AlexWayfer Now we can see some of the Checks failing, which is good, more visibility.

Don't worry, it's up to me, I'll fix it shortly (at the first glance it seems only-CI problem).

**Pull request merged**